### PR TITLE
Allow for use of later versions of PyYAML

### PR DESCRIPTION
**Pull request by candrews**
commit 5fe349a "Update PyYAML to 4.2b1 due to CVE-2017-18342" set `install_requires` to `PyYAML==4.2b1`. This exact version requirement prohibits the use of later versions of PyYAML.

### CODE REVIEW

**Comment by ioparaskev** [comment]
I'm not sure I understand what the issue with the pinned version is? The CVE is for versions before the one that is pinned so this project is not affected.

The point of pinning versions and not using `>=` is that this version was tested and found to be working fine. We can't guarantee that this will be the case with version 5 for example. By setting `>=` we allow a future version 5.0 to be installed without knowing if this is working or not. 

Since there's no CVE related to the pinned version, or any feature we would like to use in future versions (in which case we should pin to the version that introduces the feature), I'm not sure I would prefer the `>=` change

**Comment by candrews** [comment]
The exact version prevents onionrouter from working with later versions of PyYAML. So, for example, if a distro were using pyyaml 5.3.1, then onionrouter wouldn't work with it - and it should.

Best practice is for exact versions to be specified in `requirements.txt`, but version ranges be used in `setup.py` for this reason.

**Comment by candrews** [comment]
Merging this after https://github.com/ehloonion/onionrouter/pull/18 will fix the build problem raised by Travis CI

**Comment by ioparaskev** [comment]
> Best practice is for exact versions to be specified in `requirements.txt`, but version ranges be used in `setup.py` for this reason.
I agree but that means that we should set an upper range for the `PyYAML` then. 

For example, while preparing to launch one last version of onionrouter before we drop support for python 2.7 (as you MR suggests) I noticed that `dnspython` is not pinned in `setup.py`. What this means is that when installing the package, you'll fetch the latest `dnspython` package. Which at the moment is `2.0.0` and does not support python2. 
Which results in the following error (in python2.7)
```
onionrouter -d
Traceback (most recent call last):
  File ".virtualenvs/tmp-84bb89f45aecb1f/bin/onionrouter", line 7, in <module>
    from onionrouter.rerouter import main
  File ".virtualenvs/tmp-84bb89f45aecb1f/lib/python2.7/site-packages/onionrouter/rerouter.py", line 9, in <module>
    from onionrouter.lookups import OnionServiceLookup
  File ".virtualenvs/tmp-84bb89f45aecb1f/lib/python2.7/site-packages/onionrouter/lookups.py", line 1, in <module>
    from dns.resolver import Resolver
  File ".virtualenvs/tmp-84bb89f45aecb1f/lib/python2.7/site-packages/dns/resolver.py", line 19, in <module>
    from urllib.parse import urlparse
ImportError: No module named parse
```
I think this shows why we should have a range for version support.

**Comment by candrews** [comment]
Can we:
* Set a range with a min and a max version specified for PyYAML
* Release a final Python 2.7 version
* Set just a min version for PyYAML
* Drop Python 2.7 support
* Make a new release

**Comment by candrews** [comment]
Can you please merge this? @ioparaskev

File: setup.py [hunk 1]
```diff
@@ -10,8 +10,8 @@
     history = history_file.read()
 
 requirements = [
-    "dnspython<=2.0.0",
-    "PyYAML==4.2b1",
+    "dnspython",
```
Comment (ioparaskev) [inline setup.py:13]:
```suggestion
    "dnspython>=2.0.0,<3.0.0",
```

File: setup.py [hunk 2]
```diff
@@ -10,8 +10,8 @@
...
+    "PyYAML>=4.2b1",
```
Comment (ioparaskev) [inline setup.py:14]:
```suggestion
    "PyYAML>=4.2b1,<6.0.0",
```

**Comment by ioparaskev** [comment]
> Can we:
> 
> * Set a range with a min and a max version specified for PyYAML
> * Set just a min version for PyYAML

I did that as a suggestion for both PyYAML and dnspython. If you agree then I/you can apply them and merge this

> * Release a final Python 2.7 version
> * Drop Python 2.7 support

Last python 2.7 version was v0.5.4. I've created a separate branch  in case someone needs to continue support for this
Python 2.7 is dropped since the last merge

> * Make a new release

After merging this, I'll make a new release

**Review by ioparaskev — CHANGES_REQUESTED** [review]

File: setup.py (hunk 2 above)
Comment (candrews) [reply setup.py]: :+1:

File: setup.py (hunk 1 above)
Comment (candrews) [reply setup.py]: :+1:

**Comment by candrews** [comment]
I've made the requested changes - I think this look good :+1:

**Review by ioparaskev — APPROVED** [review]

**Comment by candrews** [comment]
@ioparaskev can you please release version 0.5.5?

**Comment by ioparaskev** [comment]
Just released version 0.6.0 thank you very much (also uploaded to pypi)

**Pull request merged**